src/utils/mutex.ts

```diff
@@ -0,0 +1,52 @@
+
+// This is the Mutex(Not to confuse with RW-Mutex). This is to ensure that certain access are only done
```

Comment:
I suggest to add it to the /** ... */ JSDoc comment before a class declaration. Please make the comment lines not so long